pkg/ingress/parser.go

```diff
@@ -274,7 +275,7 @@ func (c *Controller) generateConfig() error {
 			if len(eps) > 0 {
 				def := &haproxy.TCPService{
 					SharedInfo:   si,
-					FrontendName: fmt.Sprintf("tcp-%s", rule.TCP.Port),
+					FrontendName: fmt.Sprintf("tcp-%s", rule.TCP.Port.String()),
```

Comment:
Do we need to call `.String()` method? If yes, then `Ingress.IsValid()` method also needs to call String() in many places.